policy-controller/k8s/api/src/policy.rs

```diff
@@ -1,5 +1,119 @@
+pub mod authorization_policy;
+pub mod meshtls_authentication;
+pub mod network_authentication;
 pub mod server;
 pub mod server_authorization;
 
-pub use self::server::{Server, ServerSpec};
-pub use self::server_authorization::{ServerAuthorization, ServerAuthorizationSpec};
+pub use self::{
+    authorization_policy::{AuthorizationPolicy, AuthorizationPolicySpec},
+    meshtls_authentication::{MeshTLSAuthentication, MeshTLSAuthenticationSpec},
+    network_authentication::{NetworkAuthentication, NetworkAuthenticationSpec},
+    server::{Server, ServerSpec},
+    server_authorization::{ServerAuthorization, ServerAuthorizationSpec},
+};
+
+/// Targets a resource--or resource type--within a the same namespace.
+#[derive(
+    Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize, schemars::JsonSchema,
+)]
+pub struct TargetRef {
+    pub group: Option<String>,
+    pub kind: String,
+    pub name: Option<String>,
+    pub namespace: Option<String>,
+}
```

Comment:
Is there a reason to not follow the convention of the already popular: `TypedLocalObjectReference`

```
// TypedLocalObjectReference contains enough information to let you locate the
// typed referenced object inside the same namespace.
// +structType=atomic
type TypedLocalObjectReference struct {
	// APIGroup is the group for the resource being referenced.
	// If APIGroup is not specified, the specified Kind must be in the core API group.
	// For any other third-party types, APIGroup is required.
	// +optional
	APIGroup *string `json:"apiGroup" protobuf:"bytes,1,opt,name=apiGroup"`
	// Kind is the type of resource being referenced
	Kind string `json:"kind" protobuf:"bytes,2,opt,name=kind"`
	// Name is the name of resource being referenced
	Name string `json:"name" protobuf:"bytes,3,opt,name=name"`
}
```

Reply:
This is based on the 'target reference API' proposed by the Gateway API https://gateway-api.sigs.k8s.io/v1alpha2/references/policy-attachment/#target-reference-api